### PR TITLE
🐛 Fix not saving CVSS scores on created affects and in other edge cases

### PR DESCRIPTION
**Pull request by superbuggy**
# [OSIDB-3100] [Title]

## Checklist:

- [x] Linting passed
- [x] Type checks passed
- [x] Tests suite passed
- [x] Commits consolidated
- [ ] Changelog updated
- [-] Test cases added/updated
- [x] Jira ticket updated

## Summary:

Newly created Affect CVSSs were not being saved.

## Changes:

Refactors the saving logic to be clearer and cover edge cases.
- Track CVSS modifications separately from other changes in affects, saving in network calls.

## Considerations:

Saves a pointless network call if only affect CVSS scores are changed.

Closes OSIDB-3100.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 * Exclude empty state from open issues filter (`OSIDB-3067`)
 * Affect CVSS scores could not be edited (`OSIDB-3042`)
 * Some affects' trackers were not showing (`OSIDB-3065`, `OSIDB-3074`)
+* Affect CVSS scores wouldn't save properly (`OSIDB-3100`)
```
Comment (C-Valen) [inline CHANGELOG.md:13]: Should this go on new "Unreleased" block?

**Review by C-Valen — COMMENTED** [review]
I see lot of code for checking if affects were modified. I wonder why we cannot just use a full clone of the saved affects when entering into a flaw details and just do a deep equals between the ref of the affects and the saved clone.

Also I don't understand from the code what exact issue was preventing cvss scores to be saved.

**Review by MrMarble — APPROVED** [review]

**Review by C-Valen — APPROVED** [review]

**Review by tboyetteredhat — APPROVED** [review]
